hack/prow/run_tests.py

```diff
@@ -0,0 +1,140 @@
+  finished_json_file = open('%s/finished.json' % outdir, 'w')
+  junit_xml_file = open('%s/artifacts/junit_runner.xml' % outdir, 'w')
+
+  failures = 0
+  testxml = ""
+  for test in test_results:
+    testxml += '<testcase classname="%s" name="%s" time="%s">' % (test['classname'], test['name'], test['time'])
+    if test['status'] == 'FAIL':
+        failures += 1
+        testxml += '<failure message="Test Failed" />'
+    testxml += '</testcase>\n'
+  junit_xml_file.write('<testsuite failures="%s" tests="%s">\n' % (failures, len(test_results)))
+  junit_xml_file.write(testxml)
+  junit_xml_file.write('</testsuite>')
+  junit_xml_file.close()
+
+  started_json_file.write(json.dumps(started)) 
+  started_json_file.close()
+  finished_json_file.write(json.dumps(finished)) 
+  finished_json_file.close()
+
+  return
+
+def upload_results(outdir, test, buildnum, bucket):
+  """ push the contents of gcs_out/* into bucket/test/logs/buildnum"""
+  classname = os.path.basename(test).split('.')[0]
+  args = shlex.split("gsutil cp -R gcs_out/ gs://%s/logs/%s/%s" % (bucket, classname, buildnum))
+  p = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+  for line in p.stdout:
+    print line
+
+def gather_artifacts():
+  """ gather any named or default artifacts into artifacts/ """
+  return
+
+def run_tests(test, build_log, exit_status, started, finished, test_results):
+  """ execute the test, grab the start time, finish time, build logs and exit status
+      Pull test results and important information out of the build log 
+      test results format should be:
+      === RUN   TestFunctional/Mounting
+      --- PASS: TestFunctional (42.87s)
+          --- PASS: TestFunctional/Status (2.33s)
+          --- FAIL: SOMETESTSUITE/TESTNAME (seconds)
+  """
+  classname = os.path.basename(test).split('.')[0]
+  build_log_file = open(build_log, 'w')
+  args = shlex.split('bash -x %s' % test)
+  p = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+  for line in p.stdout:
+    build_log_file.write(line)
+    print line.rstrip()
+    if '--- PASS' in line:
+      match = re.match('.*--- PASS: ([^ ]+) \(([0-9.]+)s\)', line)
+      (name, seconds) = match.group(1, 2)
+      test_results.append({"name":name, "classname":classname, "time":seconds, "status":"PASS"})
+    if '--- FAIL' in line:
+      match = re.match('.*--- FAIL: ([^ ]+) \(([0-9.]+)s\)', line)
+      (name, seconds) = match.group(1, 2)
+      test_results.append({"name":name, "classname":classname, "time":seconds, "status":"FAIL"})
+  build_log_file.close()
+  return
+
+def main(argv):
+
+  parser = argparse.ArgumentParser(description='Run tests and upload results to GCS bucket', usage='./run_tests.py --test path/to/test.sh')
+  parser.add_argument('--test', required=True, help='give full path to test script you want to run')
+  parser.add_argument('--buildnum', required=True, help='buildnumber for uploading to GCS')
```

Comment:
"buildnum" is difficult for non-english speakers to understand. How about "--build-num"?